src/query/query_hash_scan.c

```diff
@@ -342,7 +342,56 @@ qdata_copy_hscan_key (cubthread::entry * thread_p, HASH_SCAN_KEY * key, REGU_VAR
 }
 
 /*
- * qdata_alloc_agg_hkey () - allocate new hash aggregate key
+ * qdata_copy_hscan_key_without_alloc () - deep copy hash key
+ *   returns: pointer to new hash key
+ *   thread_p(in): thread
+ *   key(in): source key
+ */
+HASH_SCAN_KEY *
+qdata_copy_hscan_key_without_alloc (cubthread::entry * thread_p, HASH_SCAN_KEY * key, REGU_VARIABLE_LIST probe_regu_list,
+				    HASH_SCAN_KEY * new_key)
+{
+  DB_TYPE vtype1, vtype2;
+  TP_DOMAIN_STATUS status = DOMAIN_COMPATIBLE;
+
+  if (new_key)
+    {
+      /* copy values */
+      new_key->val_count = key->val_count;
```

Comment:
is not necessary to check if **key** is null ?

Reply:
I'll fix it